### PR TITLE
1) Developed Map and Bean Enrichment Operators 2) MLHR-1697: Developed J...

### PR DESCRIPTION
**Pull request by chaithu14**
...DBC Loader 3) MLHR-1698: Developed HBase Loader

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/enrichment/AbstractEnrichmentOperator.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Base class for Enrichment Operator.&nbsp; Subclasses should provide implementation to getKey and convert.
+ * The operator receives a tuple and emits enriched tuple based on includeFields and lookupFields. <br/>
+ *
+ * Properties:<br>
+ * <b>lookupFieldsStr</b>: List of comma seperated keys for quick searching. Ex: Field1,Field2,Field3<br>
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/enrichment/AbstractEnrichmentOperator.java]: typo (seperated)

File: contrib/src/main/java/com/datatorrent/contrib/enrichment/AbstractEnrichmentOperator.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  /**
+   * Keep lookup data cache for fast access.
+   */
+  private transient CacheManager cacheManager;
+
+  private transient CacheStore primaryCache = new CacheStore();
+
+  private int entryExpiryDurationInMillis = 24 * 60 * 60 * 1000;
+  private int cacheCleanupInMillis = 24 * 60 * 60 * 1000;
+  private int cacheSize = 16 * 1024 * 1024;
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/enrichment/AbstractEnrichmentOperator.java]: Reduce default size of cache.

File: contrib/src/main/java/com/datatorrent/contrib/enrichment/AbstractEnrichmentOperator.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  @NotNull
+  protected String lookupFieldsStr;
+
+  protected String includeFieldsStr;
+
+  protected transient List<String> lookupFields = new ArrayList<String>();
+  protected transient List<String> includeFields = new ArrayList<String>();
+
+  protected void processTuple(INPUT tuple) {
+    Object result = cacheManager.get(getKey(tuple));
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/enrichment/AbstractEnrichmentOperator.java]: check for null value from getKey.

File: contrib/src/main/java/com/datatorrent/contrib/enrichment/BeanEnrichmentOperator.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+ * @category Database
+ * @tags enrichment, lookup
+ *
+ * @since 2.1.0
+ */
+public class BeanEnrichmentOperator extends AbstractEnrichmentOperator<Object, Object> {
+
+  private transient static final Logger logger = LoggerFactory.getLogger(BeanEnrichmentOperator.class);
+  public Class inputClass;
+  public Class outputClass;
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/enrichment/BeanEnrichmentOperator.java]: change inputClass and outputClass to private or protected.

File: contrib/src/main/java/com/datatorrent/contrib/enrichment/FSLoader.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+          for(String s: includeFields) {
+            includeTuple.add(tuple.get(s));
+          }
+          result.put(getKey(tuple), includeTuple);
+        } catch (JsonProcessingException parseExp) {
+          logger.info("Unable to parse line {}", line);
+        }
+      }
+      IOUtils.closeQuietly(bin);
+      IOUtils.closeQuietly(in);
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/enrichment/FSLoader.java]: Remove from here as these streams are closed in finally block..

File: contrib/src/main/java/com/datatorrent/contrib/enrichment/MapEnrichmentOperator.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
+package com.datatorrent.contrib.enrichment;
+
+import java.util.*;
```
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/enrichment/MapEnrichmentOperator.java]: import individual class

File: contrib/src/main/java/com/datatorrent/contrib/enrichment/BeanEnrichmentOperator.java [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  @Override
+  protected Object convert(Object in, Object cached) {
+    try {
+      Object o = outputClass.newInstance();
+      // Copy the fields from input to output
+      Field[] fields = inputClass.getFields();
+
+      for(Field f : fields) {
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/enrichment/BeanEnrichmentOperator.java]: This is inefficient and not flexible. Please use PojoUtils, see other examples of POJO enabled operators or talk to Prerna.

**Pull request closed without merging**